### ISSUE TITLE
Server->client connection timed out. Last seen xxxxx msec ago.

### ISSUE THREAD

**Issue by Zignixx** [citizenfx/fivem#2619]
### What happened?

Almost every evening, our players experience a timeout (at least 60% of players, often even 100%). During this time, approximately 230+ players are active.

The Server Resource Monitor shows no hanging scripts during the timeout. The profiler has no significant entries before the timeout.

Suddenly, the following entries appear:

`[ citizen-server-impl] network thread hitch warning: timer interval of 180 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 262 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 484 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 615 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 1134 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 2165 milliseconds
[ citizen-server-impl] network thread hitch warning: timer interval of 6381 milliseconds`

After this, all players are disconnected.

We have a Wireshark capture of these timeouts. It shows that the Windows Server receives packets from the clients but doesn't send anything back.

In the profiler, we only see entries for DroppedPlayer.

It appears that the FiveM network stack is no longer sending / not responding.

If needed, we can provide the Wireshark capture and the profiler data, but not publicly as they contain all IP addresses.

![Wireshark](https://github.com/citizenfx/fivem/assets/7584854/d86e8c08-0515-44bc-a4ab-e54a27819eb0)
IP 10.X.X.105 = Server IP
IP 94.x.x.76 = Client

### Expected result

No timeouts

### Reproduction steps

Starting the server about 3 1/2 hours before, let 230 players join

### Importancy

Crash

### Area(s)

FXServer

### Specific version(s)

Server 8450 windows

### Additional information

If you want the raw wireshark files / profiler, feel free to contact me so i can give it to you.

**Comment by RedneckLovinLife** [comment]
Do we know if this is a FiveM issue? I'm having the same problem after 3yrs of having my server.

**Comment by Nova-px** [comment]
We are experiencing the same issue at 800+ players and we also can't find any information that it's related to our scripts.

**Comment by M981** [comment]
We are also experiencing this same issue. Please if anyone finds information about it, or how to get to the bottom of it. Let me know. Everything optimization wise looks good.

**Comment by M981** [comment]
A bit late but seems to be an issue with FiveM. Players should put their FiveM Updatechannel on "Release".

**Comment by CursedRedSix** [comment]
has anyone found a solution for this, it's been happening on my server for about 1 month now

 Reason: Server->client connection timed out. Last seen 58810 msec ago.

**Comment by QidouS** [comment]
If anyone has any additional information to enlighten us... 😊

**Comment by februqry** [comment]
happening on mine too, if you have a fix contact me please

**Comment by M981** [comment]
@QidouS @februqry 

When we had this issue, it was a network leak on our server, and it kept happening for quite a while, like every day. If you two still need help with it, I’d definitely be happy to try and help out.

**Comment by Mankey** [comment]
> [@QidouS](https://github.com/QidouS) [@februqry](https://github.com/februqry)
> 
> When we had this issue, it was a network leak on our server, and it kept happening for quite a while, like every day. If you two still need help with it, I’d definitely be happy to try and help out.

im having this issue, think you can help?

**Comment by CursedRedSix** [comment]
I'm still having this same issue sometimes players are not even able to connect

**Comment by QidouS** [comment]
> @QidouS @februqry 
> 
> When we had this issue, it was a network leak on our server, and it kept happening for quite a while, like every day. If you two still need help with it, I’d definitely be happy to try and help out.

Yes I want help, thank you!

**Comment by Mankey** [comment]
> > [@QidouS](https://github.com/QidouS) [@februqry](https://github.com/februqry)
> > When we had this issue, it was a network leak on our server, and it kept happening for quite a while, like every day. If you two still need help with it, I’d definitely be happy to try and help out.
> 
> Yes I want help, thank you!

Sure, best way to contact? Also, thinking my issue might be a host issue but we will see.

**Comment by M981** [comment]
A lot of people at the same time haha, i'll try my best to help anyone who still has this issue. Is it possible to contact me on discord? My discord is m981

**Comment by QidouS** [comment]
> A lot of people at the same time haha, i'll try my best to help anyone who still has this issue. Is it possible to contact me on discord? My discord is m981
> 
Add, ty !

**Comment by CursedRedSix** [comment]
I can do that my discord is cursedredsix I shall contact you at sometime
next couple hours, to be honest since ic been running my server local
hosted I don't have this issue but when I run on a vps I get this issue

**Comment by mickeygeecom** [comment]
First of; this is not the place for support.

Secondly; this is most likely due to events stacking up to an overflow.

Look into your net event sizes, their rapidness and ensure players can not unintentionally spam requests for events. F.eg. a save button passing information between client and server.

**Comment by OGSOSSY** [comment]
My server is having the same issue. Any fix to this?

**Comment by czsquizer** [comment]
Hi,
Has anyone found a proper way on how to debug the source of network hitch ? I tried listening for events sent S2C, statebag updates, but that has unfortunatelly not helped.

**Comment by QidouS** [comment]
> Hi,
> Has anyone found a proper way on how to debug the source of network hitch ? I tried listening for events sent S2C, statebag updates, but that has unfortunatelly not helped.

Hi, I often do audits and optimization, I fixed a lot of statebags but it didn't really help. In fact, I understand that it's mostly the players who have a connection problem and that FiveM doesn't accept connection drops. For example, on a modern game, if you lose your connection for 5 seconds or even 2 seconds, you are not disconnected, whereas on FiveM, even 0.5s, it timeouts you. Maybe I'm wrong and there's a solution.

**Comment by czsquizer** [comment]
Hi, 
I have been also checking the statebags and clients, however I can not find any spamming, huge, frequently called events or updated statebags. I am wondering that else could that be.

**Comment by Legacy-TacticalGamingInteractive** [comment]
An update was pushed recently. Did you guys update your artifacts and its still happening to your clients?

**Comment by QidouS** [comment]
> An update was pushed recently. Did you guys update your artifacts and its still happening to your clients?

Now that you mention it, I have the impression that it has become more pronounced since the last version of the artifact published (recommended by txadmin)

**Comment by czsquizer** [comment]
I belive this issue has to come from other aspect, we see a massive network hitch warning in the console spam. I would not say it is from the artifacts themselves.
However I am unable to narrow it down from where it could be comming. But I know Client Events nor Statebags is not the issue.
I would try making an AI generated PR to add more debug to the network thread, but that would not be accepted :D

So if someone experienced could add some debug to network thread natively to the FXServer, it would be more than appreciated.

**Comment by QidouS** [comment]
> I belive this issue has to come from other aspect, we see a massive network hitch warning in the console spam. I would not say it is from the artifacts themselves.
> However I am unable to narrow it down from where it could be comming. But I know Client Events nor Statebags is not the issue.
> I would try making an AI generated PR to add more debug to the network thread, but that would not be accepted :D
> 
> So if someone experienced could add some debug to network thread natively to the FXServer, it would be more than appreciated.

Yes, you're right, I think. In my opinion, fivem has a high sensitivity to customer connection. If the player's connection is unstable, even 1 second, then timeout.

**Comment by anvu23198** [comment]
any updates on this issue?

**Comment by mickeygeecom** [comment]
This is most likely event related cause of bad coding / payload sends
